src/stream/reader/logfile.rs

```diff
@@ -206,13 +206,16 @@ impl LogFile {
                 Ok(())
             };
 
-        if let Some(task_id) = opts.task {
-            if let Some(task_info) = self.index.get(&task_id) {
-                let instance = task_info.last_instance();
-                if !instance.finished && opts.allow_unfinished {
-                    anyhow::bail!("Stream for task {} is not finished", task_id);
+        if let Some(ref array) = opts.task {
+            for task_id in array.iter() {
+                if let Some(task_info) = self.index.get(&task_id) {
+                    let instance = task_info.last_instance();
+                    if !instance.finished && opts.allow_unfinished {
```

Comment:
Btw @spirali shouldn't this be `!opts.allow_unfinished`? It seems that it was wrong before.

Reply:
You are right. May be there should be test on this:)